docs/code-sharing-with-user-frameworks.md

```diff
@@ -0,0 +1,91 @@
+Code sharing with user frameworks
+=================================
+
+Xamarin.iOS can AOT-compile assemblies into:
+
+* A static object (one static object per assembly).
+* A dynamic library (one or more dynamic libraries per assembly).
+  This is used for incremental builds (in which case it's one dynamic library per assembly).
+* A user framework (one or more assemblies per framework).
```

Comment:
It's on by default (but many conditions cause it to become automatically disabled, one reason being that there are no extensions, another being that the main app does not target iOS 8+).

I'll expand the documentation a bit about it.